pkg/grouper/package_grouper.go

```diff
@@ -53,37 +56,42 @@ func GroupByPURL(packageSources []models.PackageSource) map[string]models.Packag
 }
 
 func isLocationExtractedSuccessfully(filePosition models.FilePosition) bool {
-	return filePosition.Line.Start > 0 && filePosition.Line.End > 0 && filePosition.Column.Start > 0 && filePosition.Column.End > 0
+	return filePosition.Line.Start > 0 && filePosition.Line.End > 0 && filePosition.Column.Start > 0 && filePosition.Column.End > 0 && filePosition.Filename != ""
 }
 
-func extractPackageLocations(pkgSource models.SourceInfo, pkgInfos models.PackageInfo) models.PackageLocations {
+func removeHostPath(source models.SourceInfo, absolutePath string) string {
+	path := filepath.ToSlash(strings.TrimPrefix(absolutePath, source.HostPath))
+	if !strings.HasPrefix(source.Path, "/") {
+		path = strings.TrimPrefix(path, "/")
+	}
+
+	return path
+}
+
+func extractPackageLocations(source models.SourceInfo, pkgInfos models.PackageInfo) models.PackageLocations {
 	locations := models.PackageLocations{
 		Block: models.PackageLocation{
-			Filename:    pkgSource.Path,
+			Filename:    removeHostPath(source, pkgInfos.BlockLocation.Filename),
```

Comment:
We should take into account the ScannerActions.PathRelativeToScanDir before applying it, it is still a CLI and the flag can be set to false

Reply:
We are taking it in account "indirectly":
- The remove host path in the package grouper is based on the HostPath that is set in the main remove host path
- If the main remove host path is not called (because the CLI flag is set to false), HostPath will be empty and thus the package grouper remove host path will not alter the filaname